### PR TITLE
(1123) Add Referrer name to the top of view referral pages

### PR DESCRIPTION
**Pull request by jsrobertson**
## Context

https://trello.com/c/FyFfGdPj/1123-use-referrerusername-field-to-display-name-of-person-who-created-a-referral-m

## Changes in this PR
Use the user `referrerUsername` value on a Referral and call manage users api to get the referring users name and display it on the view referral page and sub pages.

## Screenshots of UI changes

image

## Release checklist

[Release process documentation](../doc/how-to/perform-a-release.md)

As part of our continuous deployment strategy we must ensure that this work is
ready to be released once merged.

### Pre-merge

- [ ] There are changes required to the Accredited Programmes API for this change to work...
  - [ ] ... and they have been released to production already

### Post-merge

- [ ] This adds, extends or meaningfully modifies a feature...
  - [ ] ... and I have written or updated an end-to-end test for the happy path in the [Accredited Programmes E2E repo](https://github.com/ministryofjustice/hmpps-accredited-programmes-e2e)
- [ ] This makes new expectations of the API...
  - [ ] ... and I have notified the API developer(s) of changes to the contract tests (Pact), or the API is already compliant
- [ ] [Manually approve](../doc/how-to/perform-a-release.md#releasing-to-the-preprod-environment) release to preprod
- [ ] [Manually approve](../doc/how-to/perform-a-release.md#releasing-to-the-production-environment) release to prod

### CODE REVIEW

File: server/@types/models/Referral.ts [hunk 1]
```diff
@@ -11,6 +11,7 @@ type Referral = {
   offeringId: CourseOffering['id']
   prisonNumber: Person['prisonNumber']
   referrerId: Express.User['userId']
```
Comment (Gweaton) [inline server/@types/models/Referral.ts:13]: Do we need `referrerId` anymore? Can we remove it in this PR?

**Review by Gweaton — COMMENTED** [review]
This looks good to me - I noticed in the `UserService.getUserFromUsername` function we throw an error if we get a 404 response. Now that we're fetching and displaying this in a few places, I think it would be good to stop throwing this error and displaying something like "Deactivated user with username: AB1234" as a fallback, in case a user's account no longer exists.

File: server/utils/referrals/showReferralUtils.test.ts [hunk 2]
```diff
@@ -40,13 +40,32 @@ describe('ShowReferralUtils', () => {
...
+
+        expect(ShowReferralUtils.submissionSummaryListRows(referral.submittedOn, 'Test User')).toEqual(
+          expect.arrayContaining([
+            {
+              key: { text: 'Date referred' },
+              value: { text: 'Not known' },
+            },
+          ]),
+        )
+      })
```
Comment (yndajas) [inline server/utils/referrals/showReferralUtils.test.ts:67]: Should this be a separate commit? Seems like it's testing something unrelated to the referrer name

File: server/utils/referrals/showReferralUtils.ts [hunk 3]
```diff
@@ -30,11 +31,20 @@ export default class ShowReferralUtils {
...
+    referralSubmissionDate: Referral['submittedOn'],
+    referrerName: User['name'],
+  ): Array<GovukFrontendSummaryListRowWithKeyAndValue> {
     return [
       {
         key: { text: 'Date referred' },
-        value: { text: DateUtils.govukFormattedFullDateString(referral.submittedOn) },
+        value: {
+          text: referralSubmissionDate ? DateUtils.govukFormattedFullDateString(referralSubmissionDate) : 'Not known',
+        },
```
Comment (yndajas) [inline server/utils/referrals/showReferralUtils.ts:43]: /here - should this be a separate commit?

File: server/@types/models/Referral.ts (hunk 1 above)
Comment (jsrobertson) [reply server/@types/models/Referral.ts:13]: I'll do that as a separate PR as it's an additional 9 file change.

**Review by Gweaton — APPROVED** [review]
